### PR TITLE
fix: sentry tracing and update deps

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Updated dependencies
 - new go application framework supporting the new user agent
 - update rest of dependencies to latest

Fixed Tracing/RequestID
 - Sentry tracing should now work much better (we didn't explicitly enable it previously)
 - RequestIDs should stay the same across calls, as Sentry is not dropping the transaction anymore

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [x] License file updated, if new 3rd-party dependency is introduced

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: .snyk [hunk 1]
```diff
@@ -3,49 +3,49 @@ version: v1.25.0
 # ignores vulnerabilities until expiry date; change duration by modifying expiry date
```
Comment (PeterSchafer) [inline .snyk:3]: Please check these changes, I think they are breaking the file

Comment (bastiandoetsch) [reply .snyk:3]: They are actually fixing it :). It broke IaC before, and they are automatically added from the CLI `ignore` command.

Comment (bastiandoetsch) [reply .snyk:3]:
```
basti-work :: ~/workspace/snyk-ls ‹feat/HEAD-278_implement_vulnerability_count_api_client*› » snyk iac test

Snyk Infrastructure as Code

✔ Test completed.

Issues
  No vulnerable paths were found!

Test Failures

  Failed to parse JSON file
  Path: /Users/bdoetsch/workspace/snyk-ls/.vscode/launch.json
        /Users/bdoetsch/workspace/snyk-ls/.vscode/settings.json
        /Users/bdoetsch/workspace/snyk-ls/.vscode/tasks.json
        /Users/bdoetsch/workspace/snyk-ls/licenses/github.com/hashicorp/hcl/json/parser/test-fixtures/bad_input_128.json
        /Users/bdoetsch/workspace/snyk-ls/licenses/github.com/hashicorp/hcl/json/parser/test-fixtures/bad_input_tf_8110.json

-------------------------------------------------------

Test Summary

  Organization: platform_hammerhead
  Project name: snyk/snyk-ls

✔ Files without issues: 2
✗ Files with issues: 0
  Ignored issues: 4
  Total issues: 0 [ 0 critical, 0 high, 0 medium, 0 low ]

Tip: Re-run in debug mode to see more information: DEBUG=*snyk* <COMMAND>
If the issue persists contact support@snyk.io

-------------------------------------------------------

Tip

  New: Share your test results in the Snyk Web UI with the option --report
```

Comment (bastiandoetsch) [reply .snyk:3]: Although the `null` looks weird

Comment (bastiandoetsch) [reply .snyk]: Pushed a commit to remove the nulls. But the file works :)

File: infrastructure/snyk_api/scan.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		fallthrough
+	case "gradle":
+		fallthrough
+	case "maven":
+		if !strings.Contains(packageName, ":") {
+			return nil, fmt.Errorf("API: invalid package name %s", packageName)
+		}
+		group := strings.Split(packageName, ":")[0]
+		artifactId := strings.Split(packageName, ":")[1]
+		scanPackageURL = fmt.Sprintf("/test/%s/%s/%s/%s?%s", packageManager, group, artifactId, packageVersion, utm)
```
Comment (PeterSchafer) [inline infrastructure/snyk_api/scan.go:139]: Question: Shouldn't there be a 'v1' prefix? As to be able to use 'rest' calls as well.

**Review by PeterSchafer — APPROVED** [review]

**Pull request merged**